exonum/src/events/network.rs

```diff
@@ -115,6 +116,28 @@ impl ConnectionsPool {
         self.inner.borrow_mut().len()
     }
 
+    fn get_or_create(
+        &self,
+        peer: SocketAddr,
+        network_tx: &mpsc::Sender<NetworkEvent>,
+        handle: &Handle,
+        network_config: NetworkConfiguration,
+        handshake_params: &HandshakeParams,
+    ) -> Option<Box<dyn Future<Item = mpsc::Sender<RawMessage>, Error = io::Error>>> {
+        self.get(peer)
+            .or_else(|| {
+                self.clone()
+                    .connect_to_peer(
+                        network_config,
+                        peer,
+                        network_tx.clone(),
+                        &handle,
+                        &handshake_params,
+                    )
+            })
+            .map(|conn_tx| conn_fut(Ok(conn_tx).into_future()))
+    }
+
     fn connect_to_peer(
```

Comment:
I am just curios what will happen if node will connect_to_peer where peer is itself(it can happen with peer request logic, for example). Node handler handles this situation well but I can't get how network part will handle it. When it will drop connection to itself especially if keepalive is set?

Reply:
In the current implementation, it will never be dropped. I'll fix that.